src/app/courses/courses.component.ts

```diff
@@ -111,8 +111,12 @@ export class CoursesComponent {
 
   async addCourse(courseInfo) {
     // ...is the rest syntax for object destructuring
-    await this.couchService.post(this.dbName, { ...courseInfo });
-    this.router.navigate(['/']);
+    try {
+      await this.couchService.post(this.dbName, { ...courseInfo });
+      this.router.navigate(['/']);
+    } catch (err) {
+      console.log(err);
```

Comment:
Rather than logging the error to the console maybe it would be better to so the error directly to the user?

Reply:
I think we'll need to add an error display popup/feature in the future, so I could add a comment there that'll make it easier to pick out as a todo when implementing that.

In the meantime for development getting the full error output from CouchDB will be helpful for us whenever we access the database.